Empire/Diplomacy.h

```diff
@@ -3,17 +3,18 @@
 
 #include <string>
 #include <boost/serialization/access.hpp>
-#include <GG/Enum.h>
+#include "../util/Enum.h"
 #include "../util/Export.h"
 
 
 //! diplomatic statuses
-GG_ENUM(DiplomaticStatus,
-    INVALID_DIPLOMATIC_STATUS = -1,
-    DIPLO_WAR,
-    DIPLO_PEACE,
-    DIPLO_ALLIED,
-    NUM_DIPLO_STATUSES
+FO_ENUM(
+    (DiplomaticStatus),
+    ((INVALID_DIPLOMATIC_STATUS, "INVALID_DIPLOMATIC_STATUS", -1))
```

Comment:
why do the invalid type enums need explicit string representations whenthe rest don't?

Reply:
I suppose it because 2-tuple is used to define identifier and string representation. I suppose we doesn't need it.

Reply:
> I suppose it because 2-tuple is used to define identifier and string representation.

Not sure what your point is... You can optionally include a string representation for each enum value, which you're doing here just for the invalid enum values. The question is why are you doing that?

Reply:
Fixed